HACKING.md

```diff
@@ -0,0 +1,150 @@
+
+## Table of Contents
+1. [Introduction to ROCKs](#introduction-to-rocks)
+2. [Tools](#tools)
+3. [References](#references)
+4. [Build Guide](#build-guide)
+6. [Uisng Ceph ROCKs](#using-ceph-rocks)
+
+## Introduction to ROCKs
+> A [ROCK](https://canonical-rockcraft.readthedocs-hosted.com/en/latest/explanation/rocks/) is an OCI compliant container image archive built on top of Ubuntu LTS releases. ROCKs are interoperable with OCI-compliant tools and platforms.
+
+The Ceph ROCK packages all the required ceph-binaries along with utilities, [kubectl](https://kubernetes.io/docs/reference/kubectl/), [confd](https://github.com/kelseyhightower/confd), some configuration files and scripts.
+
+## Tools
+ROCKs are built using a snap called [rockcraft](https://github.com/canonical/rockcraft). It uses [lxd](https://snapcraft.io/lxd) to pull dependencies and build an artefact completely isolated from the host system. This makes it easier for developers to work on ROCKs without polluting their host system with unwanted dependencies.
+You can install snapcraft and lxd using snap tool.
+```bash
+sudo snap install rockcraft --classic
+sudo snap install lxd
+```
+
+> **_NOTE:_**
+For a detailed how-to-use snapcraft tool guide, check-out [ROCK Docs](https://canonical-rockcraft.readthedocs-hosted.com/en/latest/)
+
+## References
+The `rockcraft` tool uses a declarative file called `rockcraft.yaml` as the blueprint to prepare the artefact. Below is a brief description of all the `parts` that make the Ceph ROCK.
+
+### Parts
+1. **[ceph](rockcraft.yaml#L26)**
+
+    Contains the related ceph and utility packages that are fetched from the ubuntu package repository for building this part.
+
+2. **[confd](rockcraft.yaml#L76)**
+
+    Confd is a configuration management system that can actively watch a consistent kv store like etcd and change config files based on templates. Used for rook based deployments.
+
+3. **[kubectl](rockcraft.yaml#L86)**
+
+    Kubernetes provides a command line tool for communicating with the cluster's control plane, using the Kubernetes API. This tool is named `kubectl`. Kubectl is built using [kubectl.go](kubectl/kubectl.go) and `go` snap.
+
+4. **[local-files](rockcraft.yaml#L93)**
+
+    Contains bash scripts, configuration files (s3cfg, ceph.defaults, confd templates, etc) that go in the artefact.
+
+5. **[ceph-container-service](rockcraft.yaml#L19)**
+
+    ROCKs use [pebble](https://github.com/canonical/pebble) as the official entrypoint of the OCI artefact. However, orchestration tools like [Rook](https://rook.io/) and [Cephadm](https://docs.ceph.com/en/latest/cephadm/) don't rely on the configured entrypoint and overwrite it when spawning containers. Hence, the pebble entrypoint is an experimental feature that has not been tested for production deployments.
```

Comment:
I wonder about this para, I feel it lacks a bit of context what `ceph-container-service` actually is?

Reply:
Added description of the service.